core/src/main/java/org/elasticsearch/cluster/routing/RoutingService.java

```diff
@@ -62,6 +64,10 @@
     private volatile boolean routingTableDirty = false;
 
     private volatile Future scheduledRoutingTableFuture;
```

Comment:
well I think it makes the code much simper if we only ahve one... for tests we can just decrease the interval for the scheduled task?